lib/new_relic/telemetry_sdk/clients/client.rb

```diff
@@ -32,6 +40,55 @@ def report item
         report_batch [[item.to_h], nil]
       end
 
+      def log_and_retry response
+        logger.error response.message
+        raise NewRelic::TelemetrySdk::RetriableServerResponseException
+      end
+
+      def log_and_retry_later response
+        wait_time = response['Retry-After'].to_i 
+        logger.error "Connection error. Retrying in #{wait_time} seconds"
+        logger.error response.message
+        sleep wait_time
+        raise NewRelic::TelemetrySdk::RetriableServerResponseException
+      end
+
+      def log_once_and_drop_data response, data
+        log_error_once response.class, response.message
+        logger.error "Connection error. Dropping data: #{data.size} points of data"
+      end
+
+      def log_and_split_payload response, data, common_attributes
+        logger.error "Payload too large. Splitting payload in half and attempting to resend."
+        logger.error response.message
+        # splits the data in half and calls report_batch for each half
+        data1, data2 = data.each_slice((data.size/2.0).round).to_a
+        report_batch [data1, common_attributes]
```

Comment:
[The spec](https://github.com/newrelic/newrelic-telemetry-sdk-specs/blob/master/communication.md#splitting-data) says if we can't split the data (presumably because we're down to one item, in this case). We need to handle that case and ideally add a unit test as well.

Slight side note: overall, the client unit tests are doing a great job documenting behavior in various scenarios, so I encourage the creation of this test to keep up that good work!

Reply:
Oh that's interesting. Is it possible to send a single item and still get a payload too large response? I agree we should handle that case even thought it seems like an unusual one. I'll get that updated

Reply:
Yeah, that scenario would presumably indicate a bug in the backend.